lib/src/main/java/uk/gov/di/data/lep/library/dto/DeathNotification/DeathRegistrationEvent.java

```diff
@@ -0,0 +1,17 @@
+package uk.gov.di.data.lep.library.dto.DeathNotification;
+
+import uk.gov.di.data.lep.library.dto.GroJsonRecord;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public record DeathRegistrationEvent(
+    IsoDate deathDate,
+    Integer deathRegistrationID,
+    DeathRegistrationUpdateReasonType deathRegistrationUpdateReason, // Only present on update
+    String freeFormatDeathDate,
+    StructuredDateTime recordUpdateTime, // Only present on update
+    StructuredDateTime deathRegistrationTime, // Only present on even
```

Comment:
I wonder if this is clearer split into two top level types, one for Event and one for Update rather than comments and hope?